packages/@lwc/engine-dom/src/custom-elements/create-custom-element.ts

```diff
@@ -49,17 +42,37 @@ class BaseUpgradableConstructor extends HTMLElement {
         // TODO [#2970]: LWC elements cannot be upgraded via new Ctor()
         // Do we want to support this? Throw an error? Currently for backwards compat it's a no-op.
     }
-}
 
-for (const [propName, callback] of entries(LIFECYCLE_CALLBACKS)) {
-    // @ts-expect-error type-mismatch
-    BaseUpgradableConstructor.prototype[propName] = function () {
-        // If the element is in the WeakMap (i.e. it's marked as native lifecycle), and if it was upgraded by LWC,
-        // then it can use native lifecycle
+    connectedCallback() {
+        if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
+            connectRootElement(this);
+        }
+    }
+    disconnectedCallback() {
         if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
-            callback(this);
+            disconnectRootElement(this);
         }
-    };
+    }
+    formAssociatedCallback(form: HTMLFormElement | null) {
+        if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
+            runFormAssociatedCallback(this, form);
+        }
+    }
+    formDisabledCallback(disabled: boolean) {
+        if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
+            runFormDisabledCallback(this, disabled);
+        }
+    }
+    formResetCallback() {
+        if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
+            runFormResetCallback(this);
+        }
+    }
+    formStateRestoreCallback(state: FormRestoreState | null, reason: FormRestoreReason) {
+        if (isTrue(nativeLifecycleElementsToUpgradedByLWC.get(this))) {
+            runFormStateRestoreCallback(this, state, reason);
+        }
+    }
```

Comment:
This is a little repetitive, but it doesn't have any impact on the `dom-light-styled-component-create-1k-different-this-change` benchmark, and I think it's easier to debug anyway.